File: tenkei90/024.py
N,K = map(int,input().split())
A = list(map(int,input().split()))
B = list(map(int,input().split()))
dif = 0
ans = 'Yes'
for i in range(N):
    dif += abs(A[i]-B[i])

if dif > K:
    ans = 'No'
elif (K-dif) % 2 != 0:
    ans = 'No'

print(ans)

File: tenkei90/024.cpp
#include <iostream>
#include <cmath>
using namespace std;

long long N,K;
long long A[1 << 18], B[1 << 18];

int main(){
    //input
    cin >> N >> K;
    for(int i=1;i<=N;i++) cin >> A[i];
    for(int i=1;i<=N;i++) cin >> B[i];

    //check difference
    long long dif = 0;
    for(int i=1;i<=N;i++) dif += abs(A[i]-B[i]);

    if(dif>K){
        cout << "No" << endl;
        return 0;
    }

    if((K-dif)%2 != 0){
        cout << "No" << endl;
        return 0;
    }

    cout << "Yes" << endl;
    return 0;
}
